[Bug Report][4.0.1] TypeScript error when using custom validation rules

### Environment
**Vuetify Version:** 4.0.1
**Vue Version:** 3.5.29
**OS:** macOS 10.15.7 (current)

### Steps to reproduce
1. View the TypeScript error generated by line 13 of App.vue

### Expected Behavior
No TypeScript errors

### Actual Behavior
This typescript error:
```
Expected 1-2 arguments, but got 0. ts-plugin(2554)
rules.d.ts(5, 52): An argument for 'err' was not provided.
```

<img width="883" height="428" alt="Image" src="https://github.com/user-attachments/assets/57c83b4c-1be6-4af3-a49a-d95a078f2bb6" />

### Reproduction Link
[https://codesandbox.io/p/devbox/j6xncq](https://codesandbox.io/p/devbox/j6xncq)

### Other comments
Apologies for providing a codesandbox link instead of a Vuetify Playground link, but I couldn't see how to add the Rules plugin in the Playground. The codesandbox example is essentially straight out of the Validation rules doc page (https://vuetifyjs.com/en/features/rules).

I'm having trouble reproducing it in the codesandbox, but in my actual project I also see this TypeScript error from the same use of a custom rule in a template:

```
error TS2722: Cannot invoke an object which is possibly 'undefined'.

72             :rules="[rules.pinCode()]"
                        ~~~~~~~~~~~~~
```

<!-- generated by vuetify-issue-helper. DO NOT REMOVE -->

--- a/packages/vuetify/src/labs/rules/rules.ts
+++ b/packages/vuetify/src/labs/rules/rules.ts
@@ -8,12 +8,10 @@ import type { ValidationProps, ValidationRule } from '@/composables/validation'
 
 export type ValidationRuleBuilderWithoutOptions = (err?: string) => ValidationRule
 export type ValidationRuleBuilderWithOptions<T> = (options: T, err?: string) => ValidationRule
-export type ValidationRuleBuilder =
-  | ValidationRuleBuilderWithoutOptions
-  | ValidationRuleBuilderWithOptions<any>
+export type CustomValidationRuleBuilder = (...args: any[]) => ValidationRule
 
 export interface RuleAliases {
-  [name: string]: ValidationRuleBuilder
+  [name: string]: CustomValidationRuleBuilder
   required: ValidationRuleBuilderWithoutOptions
   email: ValidationRuleBuilderWithoutOptions
   number: ValidationRuleBuilderWithoutOptions

